### PR TITLE
Splints require additional hits to break based on medical skill

### PR DESCRIPTION
**Pull request by Walarks**
## About The Pull Request
Gives a splint health variable to limbs. Splint health gets decreased every time a splinted limb takes damage by the amount of damage and the splint gets removed at 0 health. Applied splint health is 15 per medical skill rank. If you splint yourself, you get 15 less splint health, but the amount can't be below 0.

## Why It's Good For The Game
As it stands right now valk's autosplint feature makes it a bit too convenient over the other modules, this should bring up the bottom line, while valk still has the benefits of instant and infinite splints.

## Changelog
:cl:
balance: splints require taking a certain amount of damage to break, based on medical skill of the one applying it. Anyone with medical skill adds 15 splint health per medical skill rank. If you splint yourself, this is reduced by 15, but can't be lower than 0.
balance: removes small chance for splints to not be damaged
/:cl:

### CODE REVIEW

**Comment by Naaanii** [comment]
Good enough. Maybe up it to 3 to take into account FF and the lack of stabilization too.

**Comment by Swagile** [comment]
this is a better PR than mine mechanically and code wise, so if this gets in ill just keep splint stack and scanner speeds on my own PR

**Comment by Losenis** [comment]
Unsure about this one since it'll only lead to marines self-splinting and then later walking up to a corpsman to tell them to unsplint and splint again. Would also mean that people with nothing better to do in the round would start asking corpsmen to do it again to the same limb after a fight to replenish the stacks for the next fight.

File: code/game/objects/items/stacks/medical.dm [hunk 1]
```diff
@@ -278,6 +278,7 @@
 	icon_state = "splint"
 	amount = 5
 	max_amount = 5
+	var/applied_splint_stacks = 1
```
Comment (TiviPlus) [inline code/game/objects/items/stacks/medical.dm:281]: Autodoc

File: code/modules/organs/limbs.dm [hunk 2]
```diff
@@ -338,9 +340,12 @@
...
 
-	if(limb_status & LIMB_SPLINTED && damage > 5 && prob(50 + damage * 2.5)) //If they have it splinted, the splint won't hold.
-		remove_limb_flags(LIMB_SPLINTED)
-		to_chat(owner, "<span class='userdanger'>The splint on your [display_name] comes apart!</span>")
+	if(limb_status & LIMB_SPLINTED && damage > 10) //If they have it splinted and no splint stacks, the splint won't hold.
+		if(splint_stacks <= 0)
+			remove_limb_flags(LIMB_SPLINTED)
+			to_chat(owner, "<span class='userdanger'>The splint on your [display_name] comes apart!</span>")
+		else
+			splint_stacks -= 1
```
Comment (TiviPlus) [inline code/modules/organs/limbs.dm:348]:
```suggestion
			splint_stacks--
```

File: code/modules/organs/limbs.dm [hunk 3]
```diff
@@ -1044,7 +1049,7 @@ Note that amputating the affected organ does in fact remove the infection from t
 			new /datum/effect_system/spark_spread(owner, owner, 5, 0, TRUE, 1 SECONDS)
 
 
-/datum/limb/proc/apply_splints(obj/item/stack/medical/splint/S, mob/living/user, mob/living/carbon/human/target)
+/datum/limb/proc/apply_splints(obj/item/stack/medical/splint/S, applied_stacks, mob/living/user, mob/living/carbon/human/target)
```
Comment (TiviPlus) [inline code/modules/organs/limbs.dm:1052]: Why are you adding this as the second arg and not the last

File: code/modules/organs/limbs.dm [hunk 4]
```diff
@@ -1074,6 +1079,7 @@ Note that amputating the affected organ does in fact remove the infection from t
 		"[text1]",
 		"[text2]")
 		add_limb_flags(LIMB_SPLINTED)
+		splint_stacks = max(splint_stacks, applied_stacks)
```
Comment (TiviPlus) [inline code/modules/organs/limbs.dm:1082]: Why the max(), you already check that in line 1056?

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by SplinterGP — CHANGES_REQUESTED** [review]
why does it take amount of times damage is dealt and not damage itself ? i don't see why getting attacked 5 times with 10 damage should take your splint rather than being attacked once with 60? this implementation seems very yikes to me honestly, even more considering you made it also not reduce per hit that below 10, meaning that if a marine has tyr mk2 alot of slashes wont reduce any splint stack, making the marine close to infinite splint.

**Comment by Losenis** [comment]
We presplinting meta now

**Comment by Hydrinator** [comment]
A cool change, but needs to be implemented right. Like others said, it has to take into account the actual damage rather than number of hits.

You're otherwise just mega pissing on certain benos.

File: code/modules/organs/limbs.dm (hunk 3 above)
Comment (Walarks) [reply code/modules/organs/limbs.dm:1052]: I think it looks more ordered like this.

File: code/modules/organs/limbs.dm (hunk 4 above)
Comment (Walarks) [reply code/modules/organs/limbs.dm:1082]: removed max

File: code/game/objects/items/stacks/medical.dm (hunk 1 above)
Comment (Walarks) [reply code/game/objects/items/stacks/medical.dm]: changed

File: code/modules/organs/limbs.dm (hunk 2 above)
Comment (Walarks) [reply code/modules/organs/limbs.dm]: changed

**Review by TiviPlus — APPROVED** [review]

**Review by TiviPlus — APPROVED** [review]

**Comment by DerFlammenwerfer** [comment]
> Unsure about this one since it'll only lead to marines self-splinting and then later walking up to a corpsman to tell them to unsplint and splint again. Would also mean that people with nothing better to do in the round would start asking corpsmen to do it again to the same limb after a fight to replenish the stacks for the next fight.

Non-issue.

**Pull request merged**